Improve Build & Check CI to displays test results directly in GitHub

hientranea opened the pull request:
Issues's description: https://github.com/u-ways/kotlin-quarkus-realworld-example-app/issues/7

The workflow has passed on my fork project. You can check it here: https://github.com/hientranea/kotlin-quarkus-realworld-example-app/actions/runs/3173264532

tranphuochien committed:
Setup test reporter job

u-ways commented on .github/workflows/build.yml:
```diff
@@ -27,3 +27,21 @@ jobs:
+        with:
+          name: test-results
+          path: test-results/test/
+      - name: Display structure of downloaded files
```
No need to include this step, can it be removed please? (lines 40 and 41)

hientranea replied:
Thank you for pointing this out. I've just updated it.

tranphuochien committed:
Remove unnecessary step

u-ways commented:
Looks good now, will merge, thanks for your help :)

tranphuochien committed:
Try to fix permission problem on CI job

https://github.com/dorny/test-reporter/issues/149

u-ways commented on .github/workflows/build.yml:
```diff
@@ -29,6 +29,7 @@ jobs:
           path: build/test-results/test/
   test-reporter:
     needs: build
+    permissions: read-all|write-all
```
Can you try removing this line and add PR write permission on the top, please?

```yml
    permissions:
      pull-requests: write
```

tranphuochien committed:
Try to fix permission problem on CI job (2)

https://github.com/actions/first-interaction/issues/10#issuecomment-1232740076
https://github.com/dorny/test-reporter/issues/149#issuecomment-1004820190

tranphuochien committed:
Try to fix permission problem on CI job (3)

u-ways commented on .github/workflows/build.yml:
```diff
@@ -7,6 +7,10 @@ on:
       - main
+permissions:
+  pull-requests: write
+  contents: read
```
@hientranea I think I will merge this and raise another ticket to try and resolve this issue for new contributors PR.

hientranea replied:
Sure, I'll fix it. Anyway, it passed on my repo with the same code https://github.com/hientranea/kotlin-quarkus-realworld-example-app/actions/runs/3174494671/jobs/5171391374 😂

tranphuochien committed:
Remove unnecessary lines

u-ways commented:
@tranphuochien thanks! Sorry this ticket gave you more trouble than it should've, I think this seems to be a common issue with CI builds in GH actions. I will raise a ticket and link it here. (feel free to pick it up if interested)

hientranea commented:
Ohh TIL. Thank you 👍
